Bad_Practices/Task_Estimation/TEBP.md

```diff
@@ -0,0 +1,13 @@
+## Mistakes to avoid when performing task estimation 
+
+____
+
+### 1. Excesive Optismism
+Providing an estimation of time for how long it will take to complete tasks should be optimsitic with a certain level of confidence, however sometimes this optimism can lead to many issues. Developers tend to forget about many of the trivial parts of development process, meetings, absences, technical issues and much more.Care should be taken to not create estimates thinking in terms of _"Coding Hours"_.
+
+### 2. Disregard For Agile
+Project requirementsare almost never set in stone, there is always a chnache of the project scope changing. The customer could request another feature at a whims notice.Agile was created in order to contsantly monitor customer feedback throughtout the development process to get a good idea of where the product is heading allowing estimates to be revised frequently, making them more accurate. Not incorporating agile in your design process will lead to problems throughout the project as it will create a sense of rigidity, stopping features to be added and making estimates less accuarate.
+
+
+### 3. Proper Time Allocation
+Most software development teams would rather **under-promise** and **over-deliver** rather then **over-promise** and **under-deliver**, this is done to avoid the scenario of not developing something that was promsised. If an entire day is allocaed to a certain task, you are more then likely gping to spend the whole day completing atasj that could be done in less time due to this generous padding of time. Tasks should be broken down accurately and assessed properly in order to allocate the right amount of time
```

Comment:
```suggestion
Most software development teams would rather **under-promise** and **over-deliver** rather then **over-promise** and **under-deliver**, this is done to avoid the scenario of not developing something that was promised. If an entire day is allocated to a certain task, you are more than likely going to spend the whole day completing a task that could be done in less time due to this generous padding of time. Tasks should be broken down accurately and assessed properly in order to allocate the right amount of time
```